core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -592,7 +586,7 @@ class TransactionCoordinator(txnConfig: TransactionConfig,
 
             val isRetry = retryOnEpochBump || retryOnOverflow
 
-            def generateTxnTransitMetadataForTxnCompletion(nextState: TransactionState): ApiResult[(Int, TxnTransitMetadata)] = {
+            def generateTxnTransitMetadataForTxnCompletion(nextState: TransactionState, isEmptyAbort: Boolean): ApiResult[(Int, TxnTransitMetadata)] = {
```

Comment:
does isEmptyAbort apply to the complete states too?

Reply:
Yes, though it refers to the client trying to abort the transaction with no partition involved(empty), the name sounds misleading which meaning more like abort on "Empty" state.
Maybe call it "isNoPartitionAbort"?

Reply:
Or something like "no partitions added" ? But yeah, changing the name may make it a bit less confusing.